Readdition of SP

Tiny-DM opened the pull request:
## What type of PR is this.
1. [ ] Bug
2. [ ] Change
3. [ ] Enhancement

### What have you changed and why?
Readded SP support to the mod, but more importantly
ANTISTASI UNIT TESTS!!!!!! im gonna write an automated testing script for antistasi, im super fun to be around at parties
You can run -autotest and boot directly into a mission. See commands below
SP is WAI otherwise from brief tests. A lot of stuff that might potentially break, is hopefully tested.

### Please verify the following and ensure all checks are completed.

1. [X] Have you loaded the mission in LAN host?
2. [ ] Have you loaded the mission on a dedicated server?

### Is further testing or are further changes required?
1. [ ] No
2. [X] Yes (Please provide further detail below.)

### How can the changes be tested?
Steps: 

********************************************************
Notes: Hoping to figure out how to prune the visible list of scenarios. Not a huge deal but I was hoping to find a workaround. Also need to fix the tooltips and give a better idea of what the solo / host buttons do

-autotest code:
```
class TestMissions
{
	class TestCase01
	{
		campaign = "";
		mission = "C:\Users\Tiny\Documents\GitHub\A3-Antistasi\A3A\addons\maps\Antistasi_Altis.Altis";
	};
};
```
to set your preferred game: `profileNamespace setVariable ["A3A_preferredTestingSave", "18411"]`

Tiny-DM committed:
basic setup

jaj22 commented on A3A/addons/core/functions/Save/fn_saveLoop.sqf:
```diff
@@ -77,7 +77,7 @@ _resourcesBackground = server getVariable "resourcesFIA";
 	if ((_friendX getVariable ["spawner",false]) and (side group _friendX == teamPlayer))then {
 		if ((alive _friendX) and (!isPlayer _friendX)) then {
-			if (((isPlayer leader _friendX) and (!isMultiplayer)) or (group _friendX in (hcAllGroups theBoss)) and (not((group _friendX) getVariable ["esNATO",false]))) then {
+			if ((isPlayer leader _friendX) or (group _friendX in (hcAllGroups theBoss)) and (not((group _friendX) getVariable ["esNATO",false]))) then {
```
Pretty sure this clause (isPlayer leader _friendX) should be deleted. I think what it did was refund troops in your personal squad, but that functionality only exists because the player didn't have separate money in SP. With this change it's a double-refund.

Tiny-DM replied:
Yeah I was having trouble reading it. Will see if I can revise

jaj22 commented on A3A/addons/core/functions/Save/fn_saveLoop.sqf:
```diff
@@ -89,7 +89,7 @@ _resourcesBackground = server getVariable "resourcesFIA";
 					if (isNil {_veh getVariable "markerX"}) then {
 						if ((_veh isKindOf "StaticWeapon") or (driver _veh == _friendX)) then {
-							if ((group _friendX in (hcAllGroups theBoss)) or (!isMultiplayer)) then {
+							if ((group _friendX in (hcAllGroups theBoss))) then {
```
Brackets lol

jaj22 commented on A3A/addons/core/functions/init/fn_initClient.sqf:
```diff
@@ -175,50 +181,8 @@ if (isNil "ace_noradio_enabled" or {!ace_noradio_enabled}) then {
 player setvariable ["compromised",0];
 
-// Install the non respawn-persistent client event handlers
-call A3A_fnc_installClientEH;
```
Deleted this call but it's not done from newPlayerSetup?

jaj22 commented on A3A/addons/core/functions/init/fn_installClientEH.sqf:
```diff
@@ -1,3 +1,4 @@
+if (!isMultiplayer) then {call A3A_fnc_newPlayerSetup};
```
Yeah this is backwards

jaj22 commented on A3A/addons/gui/mainmenu.hpp:
```diff
@@ -82,7 +82,7 @@ class RscDisplayMain : RscStandardDisplay
                     tooltip = "Start an Antistasi Game";
                     y = "(5 * 	1.5) * 	(pixelH * pixelGrid * 2) + 	(pixelH)";
-					onButtonClick="finddisplay 0 createDisplay 'A3A_MapSelectorDialog';";
+					onButtonClick="playMission ['', '\A3\Missions_F_Bootcamp\Scenarios\Arsenal.VR'];";
```
Test code?

jaj22 commented on A3A/addons/core/functions/proxy/fn_onPlayerRespawn.sqf:
```diff
@@ -29,74 +17,82 @@ _newUnit setVariable ["incapacitated",false,true];
+[(selectRandom unlockedmaps),"","",(selectRandom unlockedCompasses),(selectRandom unlockedwatches),""]];
+
+if (!isPlayer (leader group player)) then {(group player) selectLeader player};
+
+call A3A_fnc_installClientEH;
+// When LAN hosting, Bohemia's Zeus module code will cause the player lose Zeus access if the body is deleted after respawning.
```
Space please

jaj22 commented on A3A/addons/core/functions/init/fn_setupMonitor.sqf:
Move to another PR. Doesn't even conflict.

jaj22 commented on A3A/addons/core/functions/Revive/fn_unconscious.sqf:
```diff
@@ -134,7 +134,7 @@ else
 };
 
-if (_isPlayer and (_unit getVariable ["respawn",false])) exitWith {};
+if (_isPlayer and ((_unit getVariable ["respawn",false]) || (_unit getVariable ["respawning",false]))) exitWith {};
```
Ugh, gonna need to dismantle this one at some point.

Tiny-DM replied:
as far as I could see "respawn" is an engine variable. We need to be prefixing this stuff :/

jaj22 replied:
Nah, it's probably just a bug and should have been "respawning" before. Doesn't look like it would have had any effect: The respawn function checks "respawning" anyway, and the rest is just operations that don't do anything on corpses.

Tiny-DM committed:
update 18.11.25

Tiny-DM committed:
Update config.cpp

Tiny-DM commented:
Split a few map changes and the autotest changes out to seperate PRs. Removed all of the remaining junk from the map selector and added support for the "Scenarios" menu

Tiny-DM committed:
remove autotest code

jaj22 reviewed (changes requested):
Looks good aside from the leftover respawn/respawning. Gonna test it before I approve. but leaving this as a note in case I lose my marbles and think I need to read it again.

jaj22 commented:
Ok, couldn't get this to work. Pressing "play solo" in the UI doesn't work because startSPGame expects a map parameter that isn't being passed from the hpp. The missions are listed in the scenario tab but don't have a play button or image for some reason. Bad links maybe.

Tiny-DM committed:
Update fn_mapSelectorDialog.sqf

Tiny-DM committed:
fixes

moved AI grinder to initServer, slightly more logical place
added HCs to kill list for AI grinder
moved handlers around for respawning and readded revive handlers

Tiny-DM committed:
Merge remote-tracking branch 'upstream/unstable' into working-sp

Tiny-DM committed:
fixed broken client state command

jaj22 reviewed (approved):
Good enough. Couple of minor SP startup issues which are probably Arma's fault. MP works as before. Should probably describe SP as "experimental" somewhere though.
